### ISSUE TITLE
No cursor when writing a status

### ISSUE THREAD

**Issue by GoogleCodeExporter** [KGIII/pino-twitter#89]
```
Since i upgraded to pino 0.1.5 I can't see the cursor while i'm writing a
new status or replying someone on the "new status" box.

I don't know if I'm the only one with this problem...
Cheers
```

Original issue reported on code.google.com by `bicode...@gmail.com` on 12 Feb 2010 at 11:10

**Comment by GoogleCodeExporter** [comment]
```
I know about this problem. It is very strange, because it depends on 
installation type 
(from package or from sources), on version of application, but not on my code. 
I don't 
know, what to do with this.
So, if anybody can help me - I will be grateful.
```

Original comment by `tro...@gmail.com` on 12 Feb 2010 at 12:17
- Changed state: **Accepted**
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
I forgot to tell you that I only installed the ppa package. Could it be a 
webkit problem?
```

Original comment by `bicode...@gmail.com` on 12 Feb 2010 at 1:41

**Comment by GoogleCodeExporter** [comment]
```
I don't have any idea.
```

Original comment by `tro...@gmail.com` on 12 Feb 2010 at 1:54

**Comment by GoogleCodeExporter** [comment]
```
I got my version of Pino from the Ubuntu PPA you've got on your homepage and 
also
notice this issue.
```

Original comment by `mez.pahlan` on 14 Feb 2010 at 12:38

**Comment by GoogleCodeExporter** [comment]
```
I have installed Pino (0.1.5) from Ubuntu ppa, and the cursor is invisible here 
also, 
(Ubuntu 9.10). I really don't know about anything spacial about programing (so 
my 
advise may sound odd) but I think accidentally you have changed any code in 
this new 
version then the previous versions, so it's not supporting this gnome function. 
I 
think soon this problem will be solved.

Good luck   
```

Original comment by `massage2...@gmail.com` on 16 Feb 2010 at 7:08

**Comment by GoogleCodeExporter** [comment]
```
Same issue. It looks like ppa build is broken somehow.
I'm not familiar with vala, but i have some experience in programming, so feel 
free
to contact me in case you need some additional information on the bug. 
```

Original comment by `rebarned@gmail.com` on 20 Feb 2010 at 9:56

**Comment by GoogleCodeExporter** [comment]
```
Hi. Excellent program (formerly used gwibber). I noticed that the problem began 
after a daily update (if I 
remember correctly), which led to the current version 0.1.5. I hope this is 
helpful
```

Original comment by `evarr...@gmail.com` on 21 Feb 2010 at 12:09

**Comment by GoogleCodeExporter** [comment]
```
@rebarned, it's not a problem of PPA build, I have the same issue with 
installation from 
sources.
```

Original comment by `tro...@gmail.com` on 21 Feb 2010 at 11:41

**Comment by GoogleCodeExporter** [comment]
```
using pino-twitter-hg from Arch linux AUR package, I encounter the same issue
```

Original comment by `geob...@gmail.com` on 21 Feb 2010 at 3:01

**Comment by GoogleCodeExporter** [comment]
```
Just packaged 0.2.0 for PCLOS... Still the issue remains. This started from 
ver.0.1.5
which required vala 0.7.10. Not sure if this is the cause as I just pkg the 
stuff?
```

Original comment by `lxga...@gmail.com` on 21 Feb 2010 at 4:34

**Comment by GoogleCodeExporter** [comment]
```
I think it is a problem in source code. On my frugalware current, if I use a 
package
for 0.2.0 or build a development version from source, cursor is missing :(
```

Original comment by `fredbez...@gmail.com` on 22 Feb 2010 at 10:28

**Comment by GoogleCodeExporter** [comment]
```
I suspect it maybe a bug in in one or some libraries. One approach to try to 
resolve
this issue would be to compare the versions of libraries used in a system where 
there
is no issue with the systems where the issue do exist.

Fedora 12 has this issue, and uses:

atk-1.28.0-1.fc12
cairo-1.8.8-3.fc12
dbus-1.2.16-9.fc12
dbus-glib-0.82-2.fc12
glib2-2.22.4-2.fc12
gtk2-2.18.6-3.fc12
libgee-0.5.0-1.fc12
libsoup-2.28.2-1.fc12
libxml2-2.7.6-1.fc12
pango-1.26.2-1.fc12
unique-1.1.6-1.fc12
webkitgtk-1.1.15.4-1.fc12
```

Original comment by `darksh...@gmail.com` on 23 Feb 2010 at 4:50

**Comment by GoogleCodeExporter** [comment]
```
Ubuntu Karmic has the issue:

libatk1.0-0 1.28.0-0ubuntu1
libc6 2.10.1-0ubuntu16
libcairo2 1.8.8-2ubuntu1.1
libdbus-1-3 1.2.16-0ubuntu9
libdbus-glib-1-2 0.80-4ubuntu1
libfontconfig1 2.6.0-1ubuntu12
libfreetype6 2.3.9-5
libgee2 0.5.0-3~kkvala1
libglib2.0-0 2.22.3-0ubuntu1
libgtk2.0-0 2.18.3-1ubuntu2.2
libnotify1 0.4.5-1ubuntu1
libnotify1-gtk2.10 
libpango1.0-0 1.26.0-1
libsoup2.4-1 2.28.1-2
libunique-1.0-0 1.1.2-1ubuntu1
libwebkit-1.0-2 1.1.15.2-1
libxml2 2.7.5.dfsg-1ubuntu1.1
```

Original comment by `manuel....@gmail.com` on 24 Feb 2010 at 10:57

**Comment by GoogleCodeExporter** [comment]
```
Issue 109 has been merged into this issue.
```

Original comment by `tro...@gmail.com` on 28 Feb 2010 at 10:34

**Comment by GoogleCodeExporter** [comment]
```
Somebody knows which was the last version or a version where the bug was not 
present?
```

Original comment by `marce...@gmail.com` on 8 Mar 2010 at 6:33

**Comment by GoogleCodeExporter** [comment]
```
I just tried version 0.1.4 and I had a cursor there.
```

Original comment by `58desi...@gmail.com` on 8 Mar 2010 at 10:06

**Comment by GoogleCodeExporter** [comment]
```
Ok, then just diffing the version 0.1.4 and the last one can reveal the culprit.
```

Original comment by `marce...@gmail.com` on 8 Mar 2010 at 12:45

**Comment by GoogleCodeExporter** [comment]
```
This issue was present in Fedora 12 since 0.1.0 version. So, I don't think it's
something we could find diffing code between 0.1.4 and current version.
```

Original comment by `darksh...@gmail.com` on 8 Mar 2010 at 5:23

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
we have serveral Linuxers encountering this issue too. I think this is why we 
can't 
active the ibus, input method, to input Chinese. 
I think the Issue 120 that I just submit is caused by this "no cursor" issue.
```

Original comment by `chanyunk...@gmail.com` on 9 Mar 2010 at 4:54

**Comment by GoogleCodeExporter** [comment]
```
actually, this bug was fixed by checking "Rounded corners" in 
Preferences->Appearance. Then I 
could see all message. 
```

Original comment by `ochir.ab...@gmail.com` on 9 Mar 2010 at 8:39

**Comment by GoogleCodeExporter** [comment]
```
Is comment 21 correct? I still couldn't see the cursor when I toggled the 
"Rounded
corners" check box. 
```

Original comment by `mez.pahlan` on 9 Mar 2010 at 9:05

**Comment by GoogleCodeExporter** [comment]
```
No, #21 is another problem that was solved.
```

Original comment by `manuel....@gmail.com` on 9 Mar 2010 at 10:22

**Comment by GoogleCodeExporter** [comment]
```
Issue 121 has been merged into this issue.
```

Original comment by `tro...@gmail.com` on 9 Mar 2010 at 11:36

**Comment by GoogleCodeExporter** [comment]
```
yes, my #21 comment is for another issue when not all message is visible
```

Original comment by `ochir.ab...@gmail.com` on 11 Mar 2010 at 1:58

**Comment by GoogleCodeExporter** [comment]
```
It is difinetly control focus management problem - textedit windows thinks that 
it
isn't focused. I think that problem somwhere in subclassing because I've made 
similar
interface from scratch and couldn't repeat the issue. Then I've got sources of 
pino
and started to "extract" controls from its boxes, to layer top. I've noticed 
that
when "new status" control placed straight on toplevel window, clicing titlebar 
of
window or tray icon fast multiple times returns a focus to the textedit 
control. But
I can't figure out the cause of that problem, unfortunately.

Also, none of pino window controls can be focuced with Tab key, this is very 
bad for
accessibility, freaky guys and violates GNOME HIG.  
```

Original comment by `asavartsov` on 25 Mar 2010 at 5:11

**Comment by GoogleCodeExporter** [comment]
```
Returned true in focus_in/out_event stop others handles.

The attached patch is for pino-0.2.1
```

Original comment by `gege2061` on 25 Mar 2010 at 6:12

Attachments:
- [issue89.patch](https://storage.googleapis.com/google-code-attachments/pino-twitter/issue-89/comment-27/issue89.patch)

**Comment by GoogleCodeExporter** [comment]
```
Holy crap! You did it! Thanks a lot, I'm very grateful for this patch.

Closed by 1a1d053318 revision;
```

Original comment by `tro...@gmail.com` on 25 Mar 2010 at 6:43
- Changed state: **Fixed**
